### ISSUE TITLE
Setting robots value by entry type - follow up

### ISSUE THREAD

**Issue by russback** [nystudio107/craft-seomatic#814]
Hi.

I thnk my last comment on this was missed as the issue was closed. However as per the video in this last comment, I'm still seeing an issue whereby the settings for one entry type get overwritten when changing another entry type:

I have a Section which has 4 Entry Types. I am trying to set 1 of them to index and others not to. However when I switch between Entry Types in the Content Settings, the change applied to 1 Entry Type appears to be applied to all Entry Types.

[This video](https://www.awesomescreenshot.com/video/2039641?key=9d11e44f82d1f27b1828c7cfdf9808ab) shows me changing the settings.

Also on the frontend when viewing an Entry that is of Entry Type Gift Listing (this is the Entry Type I want to be indexed), robots is set to none. I have checked the SEO Settings on the Entry, and the Robots field does not have a value selected so it doesn’t seem to be an override causing this.

_Originally posted by @russback in https://github.com/nystudio107/craft-seomatic/issues/791#issuecomment-743113343_

**Comment by khalwat** [comment]
Sorry if I misunderstood, I thought the issue was resolved.

I will have a look.

**Comment by masonry47** [comment]
Hi there!  

We are having the exact same issue as seen in the video above when trying to change content SEO settings per entry type.  Any ideas?  Thank you!

**Comment by khalwat** [comment]
It's on my list to get this sorted for the next release; sorry for the hassles.

**Comment by masonry47** [comment]
Appreciate it, in the mean time should something like this work to achieve the same thing in the template?  Ware playing with that without success so far.

{% if entry.type == "entryTypeHandle"%}
      {% do seomatic.tag.get("robots").content("none") %}
{% endif %}

Thanks!

**Comment by russback** [comment]
Sorry to hassle, but do you have an ETA on a fix release? We are finalising a site and will be blocked by this in the next 2 weeks.

**Comment by khalwat** [comment]
I'm going to try to get it dealt with this weekend.

**Comment by khalwat** [comment]
So I figured out what's going on here, and why I initially couldn't reproduce it.

If you have an entry with multiple entry types, and you change the settings for one of the entry types, this will be the default for all of the entry types until you individually change them.

So right now without any code fixes on my end, you should be able to go in and save the settings discretely on each entry type, and it'll work as expected. You will get per-entry type settings.

But if you never explicitly save each entry type, you can see the weird behavior in your video.

**Comment by khalwat** [comment]
@russback Did you try what I mentioned above? Just save the settings for each Entry Type in SEOmatic and it will retain the correct values for each

**Comment by russback** [comment]
@khalwat I'm not seeing that behaviour. Here's another video where I'm setting the robots for one type, then another, then flipping back to the first. Note that the settings I chose for the first type have been lost.

https://www.awesomescreenshot.com/video/3010547?key=6d98d3cc9e3c08e1569f8c1de16958bb

**Comment by khalwat** [comment]
Related? https://github.com/nystudio107/craft-seomatic/issues/794

**Comment by jmauzyk** [comment]
@khalwat yeah, looks to be the same/similar thing I've been experiencing.

**Comment by masonry47** [comment]
Hey @khalwat, we are also still experiencing the issue, same as above and yes it does seem to be related to #794.  The setting we are trying to change per entry type is "Sitemap Enabled".  We are trying to disable for one entry type and leave it enabled for the other. But if we disable for one entry type, it disables for both (and other way around).  Thanks for looking into this!

**Comment by russback** [comment]
@khalwat yes it does seem related. Any news on an update schedule? The site we saw this on is due to go live next week. It's not a blocker but it does mean that some URLs could be listed on SERPs when they shouldn't, so not ideal.

Thanks.

**Comment by khalwat** [comment]
I'm gonna get this fixed @russback -- it's turning out to be a little trickier than expected. I'm cutting a release today, and then focusing on this fix for the next version (which if all goes well, will be out in time for you next week).

**Comment by khalwat** [comment]
I'm able to reproduce this issue: https://github.com/nystudio107/craft-seomatic/issues/794

...which is separate from what you're reporting here @russback -- that issue is just that if you go to the default page, it doesn't load the correct settings, it loads whatever was changed last.

- Go to the Content SEO settings for the section type.
- The URL will not have the `typeId` URL param.
- It will display the settings for whatever the most recently changed entry type settings is
- Any changes thus will affect that entry type, not the one you're supposed to be editing

If you want to verify this yourself, when you first go to the Content SEO settings page, choose one of the Entry Types from the menu. Then make changes. Then go back to another entry type via the menu. Make changes, etc.

It _should_ be displaying and saving the correct settings as long as you first choose one of the Entry Types from the menu before you begin making changes.

If you every go to a page without the `typeId` URL param in the URL, it make not be displaying the right thing.

I'd be open to screen sharing with you to get this nailed down if you're available @russback -- ping me on the Craft Discord if you're able, at: `andrew.welch`

**Comment by khalwat** [comment]
I _think_ this might be fixed in https://github.com/nystudio107/craft-seomatic/commit/a0f6551885fcd1101044a543c7b9728409316989

You can try it now by setting your semver in your `composer.json` to look like this:
```json
    "nystudio107/craft-seomatic": "dev-develop as 3.3.37”,
```
Then do a `composer clear-cache && composer update`

The reason I'm unsure is because I am unclear if you're seeing the same problem that I addressed. Can you give it a whirl @russback ?

**Comment by khalwat** [comment]
This is fixed in https://github.com/nystudio107/craft-seomatic/releases/tag/3.3.37

Let me know if you have further issues with it

**Comment by kaitlyn-higgins** [comment]
@khalwat I'm experiencing the same issue as @masonry47 above with sitemaps. Disabling sitemaps for one entry type disables for both & vice versa - leading to either all of the entries for both types coming through, or a 404 page. 

I have ensured that I select the entry type from the dropdown before making any changes or saving it so that the typeId URL param is being populated, and confirmed that the typeId is changing when I toggle between the two entry types. 

The odd thing is (and I'm not sure if this is consistent with what the others have faced or not), when I go back into the section and check the sitemap settings, they are both correctly toggled (one on, one off) - despite the sitemap not respecting that selection.  

I'm running version 3.3.44 so it seems that somehow this bug has creeped back up between 3.3.37 and 3.3.44. 

Let me know if you need any other details, or would like me to send a screen recording so you can see the issue in action.

**Comment by khalwat** [comment]
@kaitlyn-higgins Have you tried clearing caches after making these changes?

So you're saying that the correct setting is set on a per-entry type basis, but the sitemap itself is generated regardless of entry type?

**Comment by kaitlyn-higgins** [comment]
@khalwat yep, the caches have been cleared since applying the sitemap settings. 

And that's correct. I have toggled 'Sitemap Enabled' to 'on' for Internal News entries, and 'Sitemap Enabled' to 'off' for External News entries, and saved both. If I save the Internal (sitemap on) settings last, the sitemap will be generated for both Internal and External entries. If I save the External (sitemap off) settings last, the sitemap page will return a 404 (so no entries from either entry type are being sitemapped).

**Comment by khalwat** [comment]
@kaitlyn-higgins I think this is actually a similar but different issue, so I have created a new issue here: https://github.com/nystudio107/craft-seomatic/issues/929
